Source/Dafny/Resolver.cs

```diff
@@ -2317,10 +2317,9 @@ public class ModuleBindings {
       }
     }
 
-    void RegisterByMethod(Function f) {
+    void RegisterByMethod(Function f, TopLevelDeclWithMembers cl) {
       Contract.Requires(f != null && f.ByMethodBody != null);
 
-      var cl = (TopLevelDeclWithMembers)f.EnclosingClass;
```

Comment:
Is it a bug that the `enclosingclass` of `f` doesn't match cl?

Reply:
I don't think so, as I believe that it gets filled in later. It's null at this point.